src/service/CinemaService.java

```diff
@@ -0,0 +1,179 @@
+package service;
+
+import exception.AlreadyTakenSeatException;
+import exception.NonExistingSeatException;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class CinemaService {
+
+    private final Cinema cinema;
+
+    public CinemaService() {
+        cinema = initializeCinema();
+        createSeats();
+    }
+
+    private Cinema initializeCinema() {
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("Enter the number of rows:");
+        int rows = scanner.nextInt();
+
+        System.out.println("Enter the number of seats in each row:");
+        int cols = scanner.nextInt();
+
+        System.out.println();
+
+        return new Cinema(rows, cols);
+    }
+
+    private void createSeats() {
+        String[][] seats = new String[cinema.getRows()][];
+
+        for (int i = 0; i < cinema.getRows(); i++) {
+            String[] row = new String[cinema.getColumns()];
+            Arrays.fill(row, "S");
+            seats[i] = row;
+        }
+
+        cinema.setSeats(seats);
+    }
+
+    public void showMenu() {
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            System.out.print("""
+                    1. Show the seats
+                    2. Buy a ticket
+                    3. Statistics
+                    0. Exit
+                    """);
+
+            String input = scanner.nextLine();
+            System.out.println();
+
+            switch (input) {
+                case "0" -> {
+                    return;
```

Comment:
It would be better to close Scanner at the end of the program explicitly.

Reply:
fixed